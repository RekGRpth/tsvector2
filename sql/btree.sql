CREATE SCHEMA test;
CREATE EXTENSION tsvector2 SCHEMA test;
CREATE EXTENSION tsvector2;
CREATE TABLE test.t1(a tsvector2);

CREATE OR REPLACE FUNCTION test.insert_records()
RETURNS VOID AS $$
DECLARE
	x int;
BEGIN
	FOR x IN
		SELECT i FROM generate_series(1, 1000) i
	LOOP
		INSERT INTO test.t1(a)
			SELECT array_to_tsvector2(array_agg(repeat(letter, floor(random() * count + 1)::int)))
			FROM (
				SELECT chr(i) AS letter, b AS count
				FROM generate_series(ascii('a'), ascii('z')) i
				FULL OUTER JOIN
				(SELECT b FROM generate_series(3, 12) b) t2
				ON true
			) t3;
	END LOOP;
END
$$ LANGUAGE plpgsql;

SELECT test.insert_records();
CREATE INDEX i1 ON test.t1 USING btree(a);

SET enable_seqscan = off;
EXPLAIN (COSTS OFF) SELECT * FROM test.t1 WHERE a >= 'a:1 b:2'::tsvector2;
DROP SCHEMA test CASCADE;
DROP EXTENSION tsvector2 CASCADE;
