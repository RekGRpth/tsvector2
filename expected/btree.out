CREATE SCHEMA test;
CREATE EXTENSION tsvector2 SCHEMA test;
ERROR:  extension "tsvector2" must be installed in schema "public"
CREATE EXTENSION tsvector2;
CREATE TABLE test.t1(a tsvector2);
CREATE OR REPLACE FUNCTION test.insert_records()
RETURNS VOID AS $$
DECLARE
	x int;
BEGIN
	FOR x IN
		SELECT i FROM generate_series(1, 1000) i
	LOOP
		INSERT INTO test.t1(a)
			SELECT array_to_tsvector2(array_agg(repeat(letter, floor(random() * count + 1)::int)))
			FROM (
				SELECT chr(i) AS letter, b AS count
				FROM generate_series(ascii('a'), ascii('z')) i
				FULL OUTER JOIN
				(SELECT b FROM generate_series(3, 12) b) t2
				ON true
			) t3;
	END LOOP;
END
$$ LANGUAGE plpgsql;
SELECT test.insert_records();
 insert_records 
----------------
 
(1 row)

CREATE INDEX i1 ON test.t1 USING btree(a);
SET enable_seqscan = off;
EXPLAIN (COSTS OFF) SELECT * FROM test.t1 WHERE a >= 'a:1 b:2'::tsvector2;
                       QUERY PLAN                        
---------------------------------------------------------
 Bitmap Heap Scan on t1
   Recheck Cond: (a >= '''a'':1 ''b'':2'::tsvector2)
   ->  Bitmap Index Scan on i1
         Index Cond: (a >= '''a'':1 ''b'':2'::tsvector2)
(4 rows)

DROP SCHEMA test CASCADE;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to table test.t1
drop cascades to function test.insert_records()
DROP EXTENSION tsvector2 CASCADE;
